chain/network/src/config_json.rs

```diff
@@ -197,6 +197,7 @@ pub struct Config {
 }
 
 #[derive(serde::Serialize, serde::Deserialize, Clone, Debug)]
+#[serde(default)]
```

Comment:
I think this is a bit risky. It is easy to add a flag where a default value (`false`, `0` etc.) is not a “disabled experiment”. Can we make the following change instead?

```diff
-pub experimental: ExperimentalConfig,
+pub experimental: Option<ExperimentalConfig>,
```

Reply:
Or if not, then at least annotate each field individually.

Reply:
ohh yea that is a good thing to think of, good point. Although, are we not protected by the explicit listing out of each struct field in the `impl Default for ExperimentalConfig`? Like, if I make this change:

```diff
diff --git a/chain/network/src/config_json.rs b/chain/network/src/config_json.rs
index 1225e32ae..2a327bd7f 100644
--- a/chain/network/src/config_json.rs
+++ b/chain/network/src/config_json.rs
@@ -226,6 +226,7 @@ pub struct ExperimentalConfig {
     /// See `NetworkConfig`.
     /// Fields set here will override the NetworkConfig fields.
     pub network_config_overrides: NetworkConfigOverrides,
+    pub behave_properly: bool,
 }
 
 /// Overrides values from NetworkConfig.
```

and then try to build it, I get this:

```
% cargo build -p neard
   Compiling near-network v0.0.0 (/Users/marcelo/nearcore/chain/network)
error[E0063]: missing field `behave_properly` in initializer of `ExperimentalConfig`
   --> chain/network/src/config_json.rs:251:9
    |
251 |         ExperimentalConfig {
    |         ^^^^^^^^^^^^^^^^^^ missing `behave_properly`

For more information about this error, try `rustc --explain E0063`.
error: could not compile `near-network` (lib) due to previous error
```

But maybe there is some other way this could go wrong... @nagisa wdyt?

Reply:
My concern is more that people might forget to add the field into the config file(s), which might very read 

```json
{"...": "...","experimental": {}, "...": "..." }
```

for a large majority of users.

Picking a wrong default value by itself is still a bug in implementation of course, but when `#[serde(default)]` is on top of the struct it can be a little difficult for a developer to realize that such defaulting can occur at all. Meanwhile if you had to add `#[serde(default)]` on top of the field you’d think about what it is doing there, and if you forgot to add it tests would fail with parse errors or something.

Reply:
yeah agreed makes it more obvious if the fields are the ones marked w `#[serde(default)]`. updated it, PTAL. Went with that instead of the `experimental: Option<ExperimentalConfig>` because now at least we don't have to change stuff where it's referred to, but if you can think of a reason that would be better, we could just do it that way too. 

also while we're lookin at this kind of thing, looks like `near_store::config::StoreConfig` uses this `#[serde(default)]` over the struct pattern too..